src/front/wgsl.rs

```diff
@@ -614,22 +672,9 @@ impl Parser {
                     self.scopes.pop();
                     return Ok(*handle);
                 }
-                if self.std_namespace.as_deref() == Some(word) {
-                    lexer.expect(Token::DoubleColon)?;
-                    let name = lexer.next_ident()?;
-                    let mut arguments = Vec::new();
-                    lexer.expect(Token::Paren('('))?;
-                    while !lexer.skip(Token::Paren(')')) {
-                        if !arguments.is_empty() {
-                            lexer.expect(Token::Separator(','))?;
-                        }
-                        let arg = self.parse_general_expression(lexer, ctx.reborrow())?;
-                        arguments.push(arg);
-                    }
-                    crate::Expression::Call {
-                        origin: crate::FunctionOrigin::External(name.to_owned()),
-                        arguments,
-                    }
+                if let Ok(expr) = self.parse_function_call(&mut backup, ctx.reborrow()) {
```

Comment:
why are we passing the backup here mutably? That seems suspiciously weird

Reply:
Also good catch. I tried to make this simpler, but the reason why it is so weird is that it needs access to a lexer from before the current token was taken. I could pass a `&backup` and a `&mut lexer` at the same time, if that would seem better.

Reply:
Okay, I think I fixed this by returning the new lexer from `parse_function_call`.